### ISSUE TITLE
If inactivity timer was exceeded, redirect happens on rules with bypass policy

### ISSUE THREAD

**Issue by MoeRT09** [authelia/authelia#3587]
### Description
If there is a rule for a domain, where only certain resources are protected by a one_factor policy and the rest of the domain (all other resources) are on bypass and you are logged into an Authelia session by some other means, you get redirected to the login page when refreshing a resource, where the bypass policy should be active once your session expired due to inactivity. If you load the URL again, then (without logging in) access is possible again.

### Expected Behaviour
A redirect to the login page should never happen on rules with bypass policy.

### Reproduction Steps
Note: The default policy is deny.

1. Create a rule, where a resource of a domain is protected by a one_factor rule.
2. Create a bypass rule for the domain.
3. Access a resource on the domain that is not covered by the first rule.
4. Log into Authelia by some other means (or by accessing a resource covered by the rule from 1.).
5. Wait until your session expires due to inactivity.
6. Reload the resource from 3. -> You are redirected to the login page.

####
Authelia log
```
time="2022-06-22T17:09:10+02:00" level=error msg="Error caught when verifying user authorization: User alice has been inactive for too long" method=GET path=/api/verify remote_ip=XXX.XXX.XXX.XXX stack="github.com/authelia/authelia/v4/internal/handlers/handler_verify.go:478 VerifyGET.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/authelia_context.go:53 AutheliaMiddleware.func1.1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:25 SecurityHeadersCSPNone.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:35 SecurityHeadersNoStore.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:16 SecurityHeaders.func1\ngithub.com/fasthttp/router@v1.4.9/router.go:414 (*Router).Handler\ngithub.com/valyala/fasthttp@v1.36.0/http.go:153 (*Response).StatusCode\ngithub.com/valyala/fasthttp@v1.36.0/server.go:2338 (*Server).serveConn\ngithub.com/valyala/fasthttp@v1.36.0/workerpool.go:224 (*workerPool).workerFunc\ngithub.com/valyala/fasthttp@v1.36.0/workerpool.go:196 (*workerPool).getCh.func1\nruntime/asm_amd64.s:1571 goexit"
time="2022-06-22T17:09:10+02:00" level=info msg="Access to https://app.example.com/_/test (method GET) is not authorized to user alice, responding with status code 302 with location redirect to https://auth.example.com/?rd=https%3A%2F%2Fapp.example.com%2F_%2Ftest&rm=GET" method=GET path=/api/verify remote_ip=XXX.XXX.XXX.XXX
```
Rules:
```yaml
access_control:
    default_policy: deny # Standard
    rules:
    - domain: app.example.com
        resources:
        - '^/internal([/?].*)?$'
        policy: one_factor
    - domain: app.example.com
        policy: bypass
```

**Comment by mind-ar** [comment]
Hi @MoeRT09 , as far as I can see, authentication is being done before authorization, even in cases of bypassed rules. 
A workaround might be to modify your proxy (nginx) to validate only the '/internal' location with authelia

**Comment by nightah** [comment]
@MoeRT09 can you confirm that the fix in #3599 resolves your issue?

You can use the container with the following tag: `docker pull authelia/authelia:PR3599`.

**Comment by MoeRT09** [comment]
> @MoeRT09 can you confirm that the fix in #3599 resolves your issue?
> 
> You can use the container with the following tag: `docker pull authelia/authelia:PR3599`.

Yes after inactivity you no longer get redirected to the login page on resources with `bypass` policy. So this fixes the issue. Here is the log output:
```
time="2022-07-04T11:01:04+02:00" level=warning msg="User alice has been inactive for too long" method=GET path=/api/verify remote_ip=XXX.XXX.XXX.XXX
time="2022-07-04T11:01:04+02:00" level=debug msg="Check authorization of subject username= groups= ip=XXX.XXX.XXX.XXX and object https://app.example.com/_/test (method GET)."
```
